core/arch/x86/decode.c

```diff
@@ -1179,9 +1186,11 @@ read_instruction(byte *pc, byte *orig_pc, const instr_info_t **ret_info,
     } else if (info != NULL && di->evex_encoded) {
         if (!TEST(REQUIRES_EVEX, info->flags))
             info = NULL; /* invalid encoding */
-        /* XXX i#1312: This might need checks for REQUIRES_EVEX_L_0 and
-         * REQUIRES_EVEX_LL_0.
-         */
+        else if (TEST(REQUIRES_VEX_L_0, info->flags) && TEST(PREFIX_VEX_L, di->prefixes))
+            info = NULL;
```

Comment:
nit: all 3 of the others above have a comment; this looks like it's somehow different w/o it, but it's not, right?  More readable when consistent.

Reply:
Done.